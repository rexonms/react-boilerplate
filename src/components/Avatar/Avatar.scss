.container {
  width: 40px;
  height: 40px;
  border-radius: 20px;
  overflow: hidden;
  background: rgba(0,0,0,.5);
  img {
    width: 100%;
    height: 100%;
  }
}