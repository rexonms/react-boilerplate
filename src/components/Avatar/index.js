import React from 'react';
import PropTypes from 'prop-types';

import styles from './Avatar.scss';

const propTypes = {
  /** URL of the image to be displayed */
  imageURL: PropTypes.string.isRequired,
};

const defaultProps = {};

const Avatar = props => (
  <div className={`${styles.container}`}>
    <img src={props.imageURL} />
  </div>
);

Avatar.defaultProps = defaultProps;
Avatar.propTypes = propTypes;

export default Avatar;
