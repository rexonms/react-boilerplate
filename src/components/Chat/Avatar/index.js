import React from 'react';
import PropTypes from 'prop-types';

import styles from './Avatar.scss';

const propTypes = {
  /** URL of the image to be displayed */
  imageURL: PropTypes.string.isRequired,
  /** Alt text for image */
  imageAltText: PropTypes.string.isRequired,
};

const defaultProps = {};

const Avatar = props => (
  <div className={`${styles.container}`}>
    <img src={props.imageURL} alt={props.imageAltText} />
  </div>
);

Avatar.defaultProps = defaultProps;
Avatar.propTypes = propTypes;

export default Avatar;
