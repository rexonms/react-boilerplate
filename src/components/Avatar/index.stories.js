import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, select } from '@storybook/addon-knobs/react';
import { withInfo } from '@storybook/addon-info';

import Avatar from './index';

const stories = storiesOf('Components/Avatar', module);
stories.addDecorator(withKnobs);

// Knobs as dynamic variables.
stories.addDecorator(story => (story()));

stories.add('Avatar', withInfo(`
  The avatar of the user
  ~~~js
    <Avatar imageURL={imageURL} />)
  ~~~
`)(() => {
  return (
    <Avatar imageURL='https://place-hold.it/100' />);
}));

