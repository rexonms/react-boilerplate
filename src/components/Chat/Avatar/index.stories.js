import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs } from '@storybook/addon-knobs/react';
import { withInfo } from '@storybook/addon-info';

import Avatar from './index';

const stories = storiesOf('components/Chat/Avatar', module);
const imageUrl = 'https://place-hold.it/100'
stories.addDecorator(withKnobs);
stories.addDecorator(story => (story()));
stories.add('Avatar', withInfo(`
  Component
  ~~~js
    <Avatar imageURL=imageUrl imageAltText="imageAltText"/>)
  ~~~
`)(() => {
  return (
    <Avatar
      imageURL={imageUrl}
      imageAltText="imageAltText"
    />
  );
}));
