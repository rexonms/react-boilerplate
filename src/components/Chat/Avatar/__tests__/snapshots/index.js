import React from 'react';
import { shallow } from 'enzyme';

import Avatar from '../../index';

// https://hackernoon.com/oh-snap-snapshots-unit-integration-testing-with-jest-enzyme-13cc18aecb7b
it('Renders component correctly', () => {
  const component = shallow(<Avatar imageURL="https://place-hold.it/100" />);
  expect(component).toMatchSnapshot();
});
