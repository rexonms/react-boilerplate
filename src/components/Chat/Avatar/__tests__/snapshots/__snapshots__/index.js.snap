// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Avatar
    imageURL="https://place-hold.it/100"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        src="https://place-hold.it/100"
/>,
      "className": "container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "src": "https://place-hold.it/100",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          src="https://place-hold.it/100"
/>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "src": "https://place-hold.it/100",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
